### DISCUSSION TITLE
Documentation on how to use docker image is not clear

### DISCUSSION THREAD

**Discussion by maxsudhir** [fauxauldrich/camouflage#89] [General]
Hello,

Currently not much documentation is available on how to build or Use the already published docker image (https://hub.docker.com/r/shubhendumadhukar/camouflage). 

When i tried to run the docker image available in docker hub, i am receiving the below error
C:\work\personal>docker images
REPOSITORY                           TAG                                                     IMAGE ID       CREATED         SIZE
docker/desktop-kubernetes            kubernetes-v1.21.2-cni-v0.8.5-critools-v1.17.0-debian   a502c6d66bd7   2 months ago    299MB
k8s.gcr.io/kube-apiserver            v1.21.2                                                 106ff58d4308   2 months ago    126MB
k8s.gcr.io/kube-controller-manager   v1.21.2                                                 ae24db9aa2cc   2 months ago    120MB
k8s.gcr.io/kube-scheduler            v1.21.2                                                 f917b8c8f55b   2 months ago    50.6MB
k8s.gcr.io/kube-proxy                v1.21.2                                                 a6ebd1c1ad98   2 months ago    131MB
docker/getting-started               latest                                                  083d7564d904   2 months ago    28MB
shubhendumadhukar/camouflage         latest                                                  d9dfec4a109e   3 months ago    358MB
docker/desktop-vpnkit-controller     v2.0                                                    8c2c38aa676e   3 months ago    21MB
docker/desktop-storage-provisioner   v2.0                                                    99f89471f470   4 months ago    41.9MB
k8s.gcr.io/pause                     3.4.1                                                   0f8457a4c2ec   7 months ago    683kB
k8s.gcr.io/coredns/coredns           v1.8.0                                                  296a6d5035e2   10 months ago   42.5MB
k8s.gcr.io/etcd                      3.4.13-0                                                0369cf4303ff   12 months ago   253MB

C:\work\personal>docker container run --name camo -p 5000:8080 shubhendumadhukar/camouflage:latest
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'csvtojson'
Require stack:
- /app/dist/handlebar/handleBarDefinition.js
- /app/dist/handlebar/index.js
- /app/dist/index.js
- /app/bin/camouflage.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/app/dist/handlebar/handleBarDefinition.js:16:13)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/dist/handlebar/handleBarDefinition.js',
    '/app/dist/handlebar/index.js',
    '/app/dist/index.js',
    '/app/bin/camouflage.js'
  ]
}

Do you have any plans to write a helmchart for this ?

Regards,
Sudhir

**Comment by shubhendumadhukar** [comment]
Hi @maxsudhir,

In all honesty, Camouflage is not very docker friendly at this moment, specially in a Kubernetes environment.

Since mock files are stored and read from a file system, current implementation leads to each container maintaining a local filesystem. I am working on some workarounds, for example, we can have a shared volume between multiple pods, or we can use mongo grid fs to have a centralized file system. However, I am yet to figure out and test how this would actually work in a production environment.

Open to suggestions!

Regarding the issue with current docker image, let me look into it and revert.

**Comment by shubhendumadhukar** [comment]
`docker container run --name camo -p 5000:8080 shubhendumadhukar/camouflage:latest`

This should work now @maxsudhir. You might want to mount volumes as well for mocks, protofiles and configs as required. I'll add a detailed documentation on using docker images.

**Comment by shubhendumadhukar** [comment]
[Updated Documentation](https://testinggospels.github.io/camouflage/using-with-docker/)

**Reply by maxsudhir** [reply]
Thank you so much for quick reply. I was able to launch the container and write mocks (using volumes)

**Reply by shubhendumadhukar** [reply]
Awesome. I'll have an update for you on helm/kubernetes soon.
